service/block/interface.go

```diff
@@ -2,11 +2,14 @@ package block
 
 import (
 	"context"
+
+	core "github.com/celestiaorg/celestia-core/types"
 )
 
 // Fetcher encompasses the behavior necessary to fetch new "raw" blocks.
 type Fetcher interface {
 	GetBlock(ctx context.Context, height *int64) (*RawBlock, error)
+	CommitAtHeight(ctx context.Context, height *int64) (*core.Commit, error)
```

Comment:
Wait what??? Pointers to int????

Reply:
????? what's the problem? This is how it is implemented on Core side, @Wondertan .

Reply:
https://github.com/celestiaorg/celestia-core/blob/master/rpc/client/interface.go#L81

Reply:
Well, I haven't noticed this before 🤦🏻 
Pointer to int is stupid for this case..., Pls try to avoid following tendermint code patterns, and specifically in this example, most of them are __really__ nonsense.

Reply:
Just noting that `nil` stands for latest height here: https://github.com/tendermint/tendermint/blob/03ad7d6f20d9fd8fe83d31db168f433480552e94/internal/rpc/core/env.go#L181

I don't really like that pattern either. But if in doubt, following the tendermint patterns for consistency isn't wrong per se.

Reply:
Let's keep it as it is for now